shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Era.hs

```diff
@@ -5,17 +5,21 @@
 module Cardano.Ledger.Era
   ( Era,
     Crypto,
+    ValueType,
   )
 where
 
 import qualified Cardano.Ledger.Crypto as CryptoClass
 import Data.Kind (Type)
 import Data.Typeable (Typeable)
+import qualified Shelley.Spec.Ledger.Val as ValClass
 
 class
   ( CryptoClass.Crypto (Crypto e),
-    Typeable e
+    Typeable e,
+    ValClass.Val (ValueType e)
   ) =>
   Era e
   where
   type Crypto e :: Type
+  type ValueType e :: Type
```

Comment:
I think we should define `Value` as an independent type family, not as an associated type. Why? A couple of reasons:

- Asserting the constraint here (`ValClass.Val (ValueType e)`) is a very strong assertion - it prevents us from _ever_ having a ledger which drops this constraint. If in the future we want to make some changes to this class, such a constraint would prevent us from doing so. Having this instead whenever we rely on this `Val` constraint would be much more future-proof.
- The bigger issue - `Cardano.Ledger.Era` is pretty much at the root of the dependency tree - just about every other module will depend on it. We really want to minimise the things that it depends on. For example, when you want to update the `scaledMinDeposit` calculation to be era dependent, you will now have a problem due to a cyclical module import. 

Instead, we should create a module like `Cardano.Ledger.Types` and declare the various type families we will be instantiating there:

```
type family Value e :: Type
```

This will be joined (in due course) by, for example

```
type family TxBody e :: Type
```

Reply:
*Bikeshedding* The `Type` in `ValueType` seems superfluous, since we know it's a type. Let's just stick with `Value`.

Reply:
We should not be able to have a ledger that has this constraint - Plutus scripts will rely on the operations in Val for sure

Reply:
I will rename ValueType in a next PR

Reply:
> We should not be able to have a ledger that has this constraint - Plutus scripts will rely on the operations in Val for sure

Sure. But perhaps we might end up wanting to add in new operations, that every `Val` in a future epoch additionally supports, beyond those supported here.

Reply:
Val still has to be the minimal constraint for all eras, though I think? Additional operations should go into the specific era-related package, and have functions that use it only in that package